.github/workflows/lint.yaml

```diff
@@ -0,0 +1,36 @@
+# validate any terraform file changes
+name: Terraform lint
+
+env:
+  TERRAFORM_VERSION: "1.2.5"
+  TERRAGRUNT_VERSION: "0.38.6"
+
+on:
+  push:
```

Comment:
Suggestion: it should be sufficient to only run on 'pull_request' and not on 'push'. The workflow will then be triggered when a pull request is opened and also [when new update for the pull request is pushed](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request:~:text=occurs.%20For%20example%2C-,if%20no%20activity%20types%20are%20specified%2C%20the%20workflow%20runs%20when%20a%20pull%20request%20is%20opened%20or%20reopened%20or%20when%20the%20head%20branch%20of%20the%20pull%20request%20is%20updated,-.%20For%20activity%20related)